File: RetailChainService/src/main/java/com/sba301/retailmanagement/service/StockRequestService.java
package com.sba301.retailmanagement.service;

import com.sba301.retailmanagement.dto.request.StockRequestCreateRequest;
import com.sba301.retailmanagement.dto.request.StockRequestRejectRequest;
import com.sba301.retailmanagement.dto.response.StockRequestResponse;

import java.util.List;

public interface StockRequestService {

    StockRequestResponse createRequest(StockRequestCreateRequest request);

    List<StockRequestResponse> getPendingRequests();

    List<StockRequestResponse> getStoreRequests(Long storeId);

    StockRequestResponse getRequestById(Long id);

    StockRequestResponse approveRequest(Long id);

    StockRequestResponse rejectRequest(Long id, StockRequestRejectRequest request);

    StockRequestResponse cancelRequest(Long id, String reason);
}

File: RetailChainUi/src/services/stockRequest.service.js
import { axiosPrivate } from './api/axiosClient';

const stockRequestService = {
  createRequest: async (data) => {
    return axiosPrivate.post('/stock-request', data);
  },

  getStoreRequests: async (storeId) => {
    return axiosPrivate.get(`/stock-request/store/${storeId}`);
  },

  getPendingRequests: async () => {
    return axiosPrivate.get('/stock-request/pending');
  },

  getRequestById: async (id) => {
    return axiosPrivate.get(`/stock-request/${id}`);
  },

  approveRequest: async (id) => {
    return axiosPrivate.put(`/stock-request/${id}/approve`);
  },

  rejectRequest: async (id, reason) => {
    return axiosPrivate.put(`/stock-request/${id}/reject`, { reason });
  },

  cancelRequest: async (id, reason) => {
    return axiosPrivate.put(`/stock-request/${id}/cancel?reason=${encodeURIComponent(reason)}`);
  },
};

export default stockRequestService;
